### ISSUE TITLE
Client Progressive Registration: Update Business Email Question/Answer Functionality

### ISSUE THREAD

**Issue by acshields** [appirio-tech/connect-app#4107]
### Update - Workflow

- A client user is prompted to complete progressive profile completion if they have incomplete required profile information.
- The client user's account email should be displayed, in addition to any other incomplete required profile information.
- The client user should be asked: "Is this your business email address?" and the answer will be a Yes/No buttons.
- If the client user says "Yes," no other user actions for this question are required.
- If the client user says "No," a new field should appear that states "Please provide your business email" and the answer is a free text field.
- In the database, the "Business Email" should be stored as a trait. In total, there are two email attributes: Account Email and Business Email. These may be the same, or they may differ.

**Comment by vikasrohit** [comment]
@RishiRajSahu @acshields before we implement this, we need to analyze the notification emails feature as well because once we have the separate business email, we need to send notifications (generated from Connect) to that email only. As of now it picks the login email for all sort of communication.
Also, we need to be aware of email verification process as well as we don't want users to enter invalid email address. Because we are using the same email address as of login email, we are using the standard email verification feature, we have to implement email verification feature for business email as well.

**Comment by maxceem** [comment]
Do I understand this correctly, that we should only implement this task taking into account comment from Vikas https://github.com/appirio-tech/connect-app/issues/4107#issuecomment-688662696? Looks like there is some backend job has to be done first, before we can implement this, right?

**Comment by RishiRajSahu** [comment]
@maxceem Yep, I mentioned same to the @acshields when the issue was added, she will get back to us with updates. Until then we can keep it on hold.

@acshields do we have updates around this ?

**Comment by RishiRajSahu** [comment]
@acshields @mtwomey @vikasrohit 
Meeting Notes - 
- We discussed that having another email for a user is beyond the scope of requirement, as it will unnecessarily create confusions for user to treat which email is primary source of login/email-notifications.
- We discussed on exploiting the event we send to salesforce during registration, and based on the payload data's email field maybe we can put filters on salesforce side to check the domains(valid/invalid) and create an additional flag `isValid` with each lead to distinguish the leads
- We also concluded that company url should be also sufficient to identify whether the lead is valid or not
- Open Questions - Should we send Connect traits data also to salesforce ?
